csharp/src/Ice/Connection.cs

```diff
@@ -15,31 +15,48 @@
 
 namespace ZeroC.Ice
 {
-    public enum ACMClose
+    /// <summary>Specifies the close semantics for ACM (Active Connection Management).</summary>
+    public enum AcmClose
     {
+        /// <summary>Disables automatic connection closure.</summary>
         CloseOff,
+        /// <summary>Gracefully closes a connection that has been idle for the configured timeout period.</summary>
         CloseOnIdle,
+        /// <summary>Forcefully closes a connection that has been idle for the configured timeout period, regardless of
+        /// whether the connection has pending invocations or dispatch.</summary>
         CloseOnInvocation,
+        /// <summary>Forcefully closes a connection that has been idle for the configured timeout period, but only if
+        /// the connection has pending invocations.</summary>
         CloseOnInvocationAndIdle,
+        /// <summary>Combines the behaviors of CloseOnIdle and CloseOnInvocation.</summary>
         CloseOnIdleForceful
     }
 
-    public enum ACMHeartbeat
+    /// <summary>Specifies the heartbeat semantics for ACM (Active Connection Management).</summary>
+    public enum AcmHeartbeat
     {
+        /// <summary>Disables heartbeats.</summary>
         HeartbeatOff,
+        /// <summary>Send a heartbeat at regular intervals if the connection is idle and only if there are pending
+        /// dispatch.</summary>
         HeartbeatOnDispatch,
+        /// <summary>Send a heartbeat at regular intervals when the connection is idle.</summary>
         HeartbeatOnIdle,
+        /// <summary>Send a heartbeat at regular intervals until the connection is closed.</summary>
         HeartbeatAlways
     }
 
-    [Serializable]
-    public struct ACM : IEquatable<ACM>
+    /// <summary>This struct represents the Acm (Active Connection Management) configuration.</summary>
+    public struct Acm : IEquatable<Acm>
```

Comment:
Since all the struct properties are get-only, I would mark it as readonly.

Reply:
fixed